packages/evm/src/contracts/PublicContractClient.ts

```diff
@@ -96,7 +96,7 @@ export class PublicContractClient<TAbi extends readonly unknown[]> {
       address,
       functionName,
       abi: this.abi,
-    } as WriteContractParameters<TAbi, TFunctionName>;
+    } as unknown as WriteContractParameters<TAbi, TFunctionName>;
```

Comment:
we shouldn't type cast to unknown. are you doing this just to avoid the typescript errors?